### PR TITLE
PMM-6378 Add pt-summary into pmm-client package

### PR DESCRIPTION
**Pull request by EvgeniyPatlan**

### CODE REVIEW

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1053-caad276
client docker - perconalab/pmm-client-fb:PR-1053-caad276
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1053-caad276.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1053-c3b1b8f
client docker - perconalab/pmm-client-fb:PR-1053-c3b1b8f
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1053-c3b1b8f.tar.gz

File: build/bin/build-client-binary [hunk 1]
```diff
@@ -51,6 +51,12 @@ gobuild_component() {
         tar -C /tmp/go/src/${extract_path} --strip-components=1 -zxpf \${source_dir}/${component}-*.tar.gz
         pushd /tmp/go
             export GOPATH=\$(pwd -P)
+            if [ -d /tmp/go/src/github.com/percona/percona-toolkit ]; then
+                pushd /tmp/go/src/${extract_path}
+                    go get -u github.com/golang/dep/cmd/dep
```
Comment (AlekSi) [inline build/bin/build-client-binary:56]: Please install tagged release: https://github.com/golang/dep/releases/tag/v0.5.4
Please also add a comment explaining why we add another ad-hoc step

File: build/bin/build-client-source [hunk 2]
```diff
@@ -42,6 +42,49 @@ prepare_tarball() {
     fi
 }
 
+repack_toolkit() {
```
Comment (AlekSi) [inline build/bin/build-client-source:45]: UH

File: build/bin/build-client-binary (hunk 1 above)
Comment (AlekSi) [reply build/bin/build-client-binary:56]: Let's wait for @percona-csalguero's response in Slack

Comment (AlekSi) [reply build/bin/build-client-binary:56]: @EvgeniyPatlan Let's use tagged version there and below. We should not use `go get`.

File: build/bin/build-client-source (hunk 2 above)
Comment (AlekSi) [reply build/bin/build-client-source:45]: Let's keep it for now.

**Review by AlekSi — CHANGES_REQUESTED** [review]
See comments about `dep` and `go get`

File: build/bin/build-client-binary (hunk 1 above)
Comment (EvgeniyPatlan) [reply build/bin/build-client-binary:56]: @AlekSi we use the same approach in pmm-1.x. so I just ported changes.

Comment (AlekSi) [reply build/bin/build-client-binary:56]: So? We still should change that

Comment (EvgeniyPatlan) [reply build/bin/build-client-binary]: done

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1053-45da762
client docker - perconalab/pmm-client-fb:PR-1053-45da762
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1053-45da762.tar.gz

**Comment by rnovikovP** [comment]
@EvgeniyPatlan , @AlekSi  I see we are adding all other pt-*-summary tools. What about https://www.percona.com/doc/percona-toolkit/3.0/pt-pg-summary.html ? is it lost or not added intentionally?

**Comment by EvgeniyPatlan** [comment]
> @EvgeniyPatlan , @AlekSi I see we are adding all other pt-*-summary tools. What about https://www.percona.com/doc/percona-toolkit/3.0/pt-pg-summary.html ? is it lost or not added intentionally?

@rnovikovP added

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1053-fb9a1b9
client docker - perconalab/pmm-client-fb:PR-1053-fb9a1b9
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1053-fb9a1b9.tar.gz

File: build/debian/rules [hunk 3]
```diff
@@ -58,6 +58,9 @@ override_dh_auto_install:
 	cp -f distro/example.prom $(TMP)/example.prom
 	cp -f distro/queries-mysqld.yml $(TMP)/queries-mysqld.yml
 	cp -f distro/queries-postgres.yml $(TMP)/queries-postgres.yml
+	cp -f distro/pt-summary $(TMP)/pt-summary
+	cp -f distro/pt-mysql-summary $(TMP)/pt-mysql-summary
+	cp -f distro/pt-mongodb-summary $(TMP)/pt-mongodb-summary
```
Comment (AlekSi) [inline build/debian/rules:63]: pt-pg-summary there too?

Comment (EvgeniyPatlan) [reply build/debian/rules:63]: fixed

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1053-87459f4
client docker - perconalab/pmm-client-fb:PR-1053-87459f4
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1053-87459f4.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1053-87459f4
client docker - perconalab/pmm-client-fb:PR-1053-87459f4
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1053-87459f4.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1053-87459f4
client docker - perconalab/pmm-client-fb:PR-1053-87459f4
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1053-87459f4.tar.gz

**Review by AlekSi — CHANGES_REQUESTED** [review]
`sources/percona-toolkit/src/github.com/percona/percona-toolkit` is updated in that PR, but there are not changes in `.gitmodules`. Either submodule update should be reverted, or .gitmodules changed

**Comment by EvgeniyPatlan** [comment]
> `sources/percona-toolkit/src/github.com/percona/percona-toolkit` is updated in that PR, but there are not changes in `.gitmodules`. Either submodule update should be reverted, or .gitmodules changed

strange it was updated but didn't showed in git. I've updated it one more time

File: .gitmodules [hunk 4]
```diff
@@ -45,7 +45,7 @@
 [submodule "percona-toolkit"]
 	path = sources/percona-toolkit/src/github.com/percona/percona-toolkit
 	url = https://github.com/percona/percona-toolkit.git
-	branch = release-3.0.10
+	branch = release-3.2.1
```
Comment (AlekSi) [inline .gitmodules:48]: @percona-csalguero @BupycHuk Is it safe to use? Should we update our dependencies?

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1053-6c8feb4
client docker - perconalab/pmm-client-fb:PR-1053-6c8feb4
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1053-6c8feb4.tar.gz

File: build/debian/rules [hunk 5]
```diff
@@ -58,6 +58,10 @@ override_dh_auto_install:
...
+	cp -f distro/pt-pg-summary $(TMP)/pt-mongodb-summary
```
Comment (AlekSi) [inline build/debian/rules:64]: Nope :) still mongo

**Review by AlekSi — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-1053-60a9594
client docker - perconalab/pmm-client-fb:PR-1053-60a9594
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-1053-60a9594.tar.gz

File: .gitmodules (hunk 4 above)
Comment (percona-csalguero) [reply .gitmodules:48]: Yes, why not?

**Review by alexanderblinkov-percona — APPROVED** [review]

File: .gitmodules (hunk 4 above)
Comment (AlekSi) [reply .gitmodules:48]: Should we update it there: https://github.com/percona/pmm-agent/blob/adfe6ceefa5cd8aab029efcdf6c71f1d6e748a76/Gopkg.toml#L39-L42 ? Somewhere else?

File: build/debian/rules (hunk 5 above)
Comment (EvgeniyPatlan) [reply build/debian/rules]: fixed

File: .gitmodules (hunk 4 above)
Comment (AlekSi) [reply .gitmodules:48]: https://github.com/percona/pmm-agent/pull/182

**Review by AlekSi — APPROVED** [review]

**Pull request merged**